src/python_testing/TC_DeviceBasicComposition.py

```diff
@@ -775,7 +775,17 @@ def test_TC_IDM_12_1(self):
         software_version = self.endpoints[0][Clusters.BasicInformation][Clusters.BasicInformation.Attributes.SoftwareVersion]
         filename = f'device_dump_0x{vid:04X}_0x{pid:04X}_{software_version}.json'
         dump_device_composition_path = self.user_params.get("dump_device_composition_path", filename)
-        self.dump_wildcard(dump_device_composition_path)
+        json_str, txt_str = self.dump_wildcard(dump_device_composition_path)
+
+        # Structured dump so we can pull these back out of the logs
+        def log_structured_data(start_tag: str, dump_string):
+            lines = dump_string.splitlines(keepends=True)
+            logging.info(f'{start_tag}BEGIN ({len(lines)} lines)====')
+            logging.info(f'{start_tag}{start_tag.join(lines)}')
```

Comment:
There is double start-tag here which is a bit confusing.

I have an alternative:

```
[logging.info(f'{start_tag}{line}' for line in dump_string_splitlines()]
```